### PR TITLE
Updating Boot code with PHP 8.1 features

### PR DESCRIPTION
**Pull request by msmakouz**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ✔️ 
| New feature?  | ❌ 
| Issues        | #512

### CODE REVIEW

File: src/Boot/src/BootloadManager.php [hunk 1]
```diff
@@ -22,15 +15,13 @@
...
     /** @var array<class-string> */
-    private $classes = [];
+    private array $classes = [];
 
-    public function __construct(Container $container)
-    {
-        $this->container = $container;
+    public function __construct(
+        /* @internal */
+        protected Container $container
```
Comment (roxblnfk) [inline src/Boot/src/BootloadManager.php:23]: readonly?

**Review by roxblnfk — COMMENTED** [review]
\Spiral\Boot\AbstractKernel::init should be elliminated

File: src/Boot/src/BootloadManager.php (hunk 1 above)
Comment (msmakouz) [reply src/Boot/src/BootloadManager.php]: Removed AbstractKernel::init, `protected Container $container` changed to `private readonly Container $container`. This class is final and protected doesn't make sense.

File: src/Boot/src/AbstractKernel.php [hunk 2]
```diff
@@ -190,26 +177,27 @@ public function started(Closure ...$callbacks): void
...
      */
-    public function serve()
+    public function serve(): mixed
     {
         foreach ($this->dispatchers as $dispatcher) {
             if ($dispatcher->canServe()) {
                 return $this->container->runScope(
                     [DispatcherInterface::class => $dispatcher],
-                    [$dispatcher, 'serve']
+                    fn () => $dispatcher->serve()
```
[the remark this answers is not in the document]
Comment (msmakouz) [reply src/Boot/src/AbstractKernel.php]: Added static

File: src/Boot/src/FinalizerInterface.php [hunk 3]
```diff
@@ -20,12 +13,12 @@ interface FinalizerInterface
      * Finalizers are executed after every request and used for garbage collection
      * or to close open connections.
      */
-    public function addFinalizer(callable $finalizer);
+    public function addFinalizer(callable $finalizer): self;
```
[the remark this answers is not in the document]
Comment (msmakouz) [reply src/Boot/src/FinalizerInterface.php]: Added `static`, instead of `self`

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**